src/plugins/Polls/Services.tsx

```diff
@@ -0,0 +1,187 @@
+                results: [0, 0],
+            }
+            if (data.options) {
+                gun.get('polls')
+                    .get(key)
+                    .get('options')
+                    .on((options) => {
+                        delete options._
+                        poll.options = Object.values(options)
+                    })
+            }
+            if (data.results) {
+                gun.get('polls')
+                    .get(key)
+                    .get('results')
+                    .on((results) => {
+                        delete results._
+                        poll.results = Object.values(results)
+                    })
+            }
+            polls.push(poll)
+        })
+
+    return polls
+}
+
+interface voteProps {
+    poll: PollGunDB
+    index: number
+}
+
+export async function vote(props: voteProps) {
+    const { poll, index } = props
+    let results: Array<number> = [0, 0]
+    gun.get('polls')
+        .get(poll.key)
+        .get('results')
+        .on((item) => {
+            delete item._
+            results = Object.values(item)
+        })
+    const count = results[index] + 1
+    const newResults = {
+        ...results,
+        [index]: count,
+    }
+
+    gun.get('polls')
+        .get(poll.key)
+        .get('results')
+        // @ts-ignore
+        .put(newResults)
+
+    PluginMessageCenter.emit('maskbook.polls.update', undefined)
+
+    return {
+        ...poll,
+        results: Object.values(newResults),
+    }
+}
+
+export async function getPollByKey(props: { key: string | number | symbol }) {
+    const keys = (props.key as string).split('_')
+    let poll: PollGunDB = {
+        key: props.key,
+        id: keys[0],
+        sender: '',
+        question: '',
+        start_time: 0,
+        end_time: 0,
+        options: ['', ''],
+        results: [0, 0],
+    }
+    gun.get('polls')
+        .get(props.key)
+        .on((data) => {
+            poll = {
+                key: props.key,
+                id: keys[0],
+                sender: data.sender,
+                question: data.question,
+                start_time: data.start_time,
+                end_time: data.end_time,
+                options: ['', ''],
+                results: [0, 0],
+            }
+            if (data.options) {
+                gun.get('polls')
+                    .get(props.key)
+                    .get('options')
+                    .on((options) => {
+                        delete options._
+                        poll.options = Object.values(options)
+                    })
+            }
+            if (data.results) {
+                gun.get('polls')
+                    .get(props.key)
+                    .get('results')
+                    .on((results) => {
+                        delete results._
+                        poll.results = Object.values(results)
+                    })
+            }
+        })
+    return poll
+}
```

Comment:
I have seen some duplicated patterns including some magic gun operations and initializing a Poll information object. You should follow the DRY and make some abstractions in it.